pkg/agent/sysadvisor/plugin/qosaware/resource/cpu/advisor.go

```diff
@@ -144,26 +145,32 @@ func (cra *cpuResourceAdvisor) GetHeadroom() (resource.Quantity, error) {
 		return *resource.NewQuantity(int64(cra.metaServer.NumCPUs-reservePoolSize), resource.DecimalSI), nil
 	}
 
-	hasShareRegion := false
+	shareRegionRequirement := 0
+
 	totalHeadroom := resource.NewQuantity(0, resource.DecimalSI)
 	for _, r := range cra.regionMap {
+		if r.Type() == types.QoSRegionTypeShare {
+			controlKnob, err := r.GetProvision()
+			if err != nil {
+				return resource.Quantity{}, fmt.Errorf("get provision with error: %v", err)
+			}
+			shareRegionRequirement += int(controlKnob[types.ControlKnobNonReclaimedCPUSetSize].Value)
+			continue
+		}
 		headroom, err := r.GetHeadroom()
 		if err != nil {
 			return headroom, err
 		}
-		if r.Type() == types.QoSRegionTypeShare {
-			hasShareRegion = true
-		}
 		// FIXME: is it reasonable to simply add headroom together?
 		totalHeadroom.Add(headroom)
 	}
 
 	// Add headroom of numas without numa binding pods if there is no share region
-	if !hasShareRegion {
-		reservePoolSizeOfNonBindingNumas := int64(math.Ceil(float64(reservePoolSize*cra.nonBindingNumas.Size()) / float64(cra.metaServer.NumNUMANodes)))
-		headroomOfNonBindingNumas := resource.NewQuantity(int64(cra.nonBindingNumas.Size()*cra.metaServer.CPUsPerNuma())-reservePoolSizeOfNonBindingNumas, resource.DecimalSI)
-		totalHeadroom.Add(*headroomOfNonBindingNumas)
-	}
+	reservePoolSizeOfNonBindingNumas := int(math.Ceil(float64(reservePoolSize*cra.nonBindingNumas.Size()) / float64(cra.metaServer.NumNUMANodes)))
```

Comment:
reserve-size won't be adjusted during running process, so why don't we pre-calculate it, and store somewhere?

Reply:
Although the size of the  pool is fixed, nonBindingNumas is dynamically changing, so we should maintain the logic here.